extensions/pkg/terraformer/config.go

```diff
@@ -245,17 +246,29 @@ func (t *terraformer) CleanupConfiguration(ctx context.Context) error {
 	t.logger.Info("Cleaning up all terraformer configuration")
 
 	t.logger.V(1).Info("Deleting Terraform variables Secret", "name", t.variablesName)
-	if err := t.client.Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}); err != nil && !apierrors.IsNotFound(err) {
+	secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}
+	if err := t.client.Delete(ctx, secret); client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if err := extensionscontroller.DeleteAllFinalizers(ctx, t.client, secret); client.IgnoreNotFound(err) != nil {
```

Comment:
Why all finalizers?
You can remove just `TerraformerFinalizer` that we are aware is set. Removing any other finalizer can brake other controller that gardener is now aware of.

Reply:
Yep, should be similar to what we're doing for the worker controller and the mcm finalizers there.